Failed to install stack on Ubuntu 18.04.2

I Tried to follow the instructions of stack [documentation](https://docs.haskellstack.org/en/stable/README/) 
Found that stack fails on `setup` step after the installation.

> Ubuntu version: Ubuntu 18.04.2 LTS

### Steps to reproduce
1. Install stack
`$ curl -sSL https://get.haskellstack.org/ | sh `
2. start new project
`$ stack new my-project`
3. navigate to project directory
`$ cd my-project`
4. run setup step
`$ stack setup`

### Expected
`setup` step is completed successfully

### Actual
stack throws an error on `setup` step:
```
$  stack setup --verbose
Version 1.9.3, Git revision 40cf7b37526b86d1676da82167ea8758a854953b (6211 commits) x86_64 hpack-0.31.1
2019-05-11 09:11:45.371763: [debug] Checking for project config at: /home/pavel/nate/haskell/my-project/stack.yaml
2019-05-11 09:11:45.371909: [debug] Loading project config file stack.yaml
2019-05-11 09:11:45.373045: [debug] Decoding build plan from: /home/pavel/.stack/build-plan/lts-13.20.yaml
2019-05-11 09:11:45.373098: [debug] Trying to decode /home/pavel/.stack/build-plan-cache/lts-13.20.cache
2019-05-11 09:11:45.375783: [debug] Success decoding /home/pavel/.stack/build-plan-cache/lts-13.20.cache
2019-05-11 09:11:45.376029: [debug] Run process: /sbin/ldconfig -p
2019-05-11 09:11:45.380010: [debug] Process finished in 4ms: /sbin/ldconfig -p
2019-05-11 09:11:45.380218: [debug] Found shared library libtinfo.so.5 in 'ldconfig -p' output
2019-05-11 09:11:45.380553: [debug] Did not find shared library libtinfo.so.6
2019-05-11 09:11:45.380632: [debug] Did not find shared library libncursesw.so.6
2019-05-11 09:11:45.380687: [debug] Found shared library libgmp.so.10 in 'ldconfig -p' output
2019-05-11 09:11:45.380758: [debug] Did not find shared library libgmp.so.3
2019-05-11 09:11:45.380787: [debug] Potential GHC builds: standard
2019-05-11 09:11:45.380820: [debug] Found already installed GHC builds: standard
2019-05-11 09:11:45.381365: [debug] Performing a sanity check on: /home/pavel/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc
2019-05-11 09:11:45.381456: [debug] Run process within /tmp/stack-sanity-check14479/: /home/pavel/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc /tmp/stack-sanity-check14479/Main.hs -no-user-package-db
2019-05-11 09:11:45.843495: [debug] Process finished in 462ms: /home/pavel/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc /tmp/stack-sanity-check14479/Main.hs -no-user-package-db
The GHC located at /home/pavel/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc failed to compile a sanity check. Please see:

    http://docs.haskellstack.org/en/stable/install_and_upgrade/

for more information. Exception was:
Received ExitFailure 1 when running
Raw command: /home/pavel/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc /tmp/stack-sanity-check14479/Main.hs -no-user-package-db
Run from: /tmp/stack-sanity-check14479/
Standard output:

[1 of 1] Compiling Main             ( /tmp/stack-sanity-check14479/Main.hs, /tmp/stack-sanity-check14479/Main.o )
Linking /tmp/stack-sanity-check14479/Main ...
Standard error:

/usr/bin/ld.gold: error: cannot find -lgmp
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
```

### Stack version

```
$ stack --version
Version 1.9.1, Git revision f9d0042c141660e1d38f797e1d426be4a99b2a3c (6168 commits) x86_64 hpack-0.31.0
```
Looks like it failed to find `libtinfo.so.6/3` dependency?

--- a/etc/scripts/get-stack.sh
+++ b/etc/scripts/get-stack.sh
@@ -612,7 +612,7 @@ try_install_pkgs() {
 apt_get_install_pkgs() {
   missing=
   for pkg in $*; do
-    if ! dpkg -s $pkg 2>/dev/null |grep Status >/dev/null; then
+    if ! dpkg -s $pkg 2>/dev/null |grep '^Status:.*installed' >/dev/null; then
       missing="$missing $pkg"
     fi
   done

--- a/etc/scripts/get-stack.sh
+++ b/etc/scripts/get-stack.sh
@@ -610,9 +610,15 @@ try_install_pkgs() {
 
 # Install packages using apt-get
 apt_get_install_pkgs() {
-  if ! dpkg-query -W "$@"|grep -v '^\S\+\s\+.\+$' > /dev/null; then
+  missing=
+  for pkg in $*; do
+    if ! dpkg -s $pkg 2>/dev/null |grep Status >/dev/null; then
+      missing="$missing $pkg"
+    fi
+  done
+  if [ "$missing" = "" ]; then
     info "Already installed!"
-  elif ! sudocmd "install required system dependencies" apt-get install -y ${QUIET:+-qq} "$@"; then
+  elif ! sudocmd "install required system dependencies" apt-get install -y ${QUIET:+-qq}$missing; then
     die "Installing apt packages failed.  Please run 'apt-get update' and try again."
   fi
 }

